src/api/app/components/watchlist_component.rb

```diff
@@ -0,0 +1,63 @@
+class WatchlistComponent < ApplicationComponent
+  def initialize(user:)
+    super
+
+    @user = user
+  end
+
+  def watched_item?
```

Comment:
Shouldn't be this plural?

Reply:
This method is asking if the item on the current page (`current_page_item`) is already on the watchlist.

Reply:
Then maybe It would be `item_watched?` instead?

Reply:
I renamed the methods related to the item we have on current page. All of them are `current_page_item*`. So this one is renamed to  `current_page_item_in_watchlist?`.